node_modules/col-core/lib/email.js

```diff
@@ -0,0 +1,183 @@
+/**
+ * Copyright ©2015. The Regents of the University of California (Regents). All Rights Reserved.
+ *
+ * Permission to use, copy, modify, and distribute this software and its documentation
+ * for educational, research, and not-for-profit purposes, without fee and without a
+ * signed licensing agreement, is hereby granted, provided that the above copyright
+ * notice, this paragraph and the following two paragraphs appear in all copies,
+ * modifications, and distributions.
+ *
+ * Contact The Office of Technology Licensing, UC Berkeley, 2150 Shattuck Avenue,
+ * Suite 510, Berkeley, CA 94720-1620, (510) 643-7201, otl@berkeley.edu,
+ * http://ipira.berkeley.edu/industry-info for commercial licensing opportunities.
+ *
+ * IN NO EVENT SHALL REGENTS BE LIABLE TO ANY PARTY FOR DIRECT, INDIRECT, SPECIAL,
+ * INCIDENTAL, OR CONSEQUENTIAL DAMAGES, INCLUDING LOST PROFITS, ARISING OUT OF
+ * THE USE OF THIS SOFTWARE AND ITS DOCUMENTATION, EVEN IF REGENTS HAS BEEN ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * REGENTS SPECIFICALLY DISCLAIMS ANY WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE. THE
+ * SOFTWARE AND ACCOMPANYING DOCUMENTATION, IF ANY, PROVIDED HEREUNDER IS PROVIDED
+ * "AS IS". REGENTS HAS NO OBLIGATION TO PROVIDE MAINTENANCE, SUPPORT, UPDATES,
+ * ENHANCEMENTS, OR MODIFICATIONS.
+ */
+
+var _ = require('lodash');
+var config = require('config');
+var ejs = require('ejs');
+var EventEmitter = require('events');
+var EmailTemplate = require('email-templates').EmailTemplate
+var fs = require('fs');
+var JoiUri = require('joi/lib/string/uri');
+var path = require('path');
+var moment = require('moment-timezone');
+var sendgrid = require('sendgrid');
+
+var log = require('col-core/lib/logger')('col-core/email');
+
+// Used to cache the compiled email templates
+var cachedEmailTemplates = {};
+
+var emailEnabled = config.get('email.enabled');
+var sendgridClient = sendgrid(config.get('email.apiKey'));
+
+// When in debug mode (tests/development), we will emit emails
+var emitter = module.exports = new EventEmitter();
+
+var EVENT_NAMES = module.exports.EVENT_NAMES = {
+  'EMAIL_SENT': 'emailSent'
+};
+
+
+/**
+ * Send an email to a user.
+ *
+ * The email will be sent with the following from headers:
+ *  - from email address: noreply@<the course' canvas instance api domain>
+ *  - from name: The course canvas instance name or "The Asset Library and Whiteboards" if there's no configured name
```

Comment:
It might be better to make this the name of the course instead of the name of the Canvas instance? There's more courses using it now.